src/metabase/automagic_dashboards/core.clj

```diff
@@ -0,0 +1,568 @@
+                                   :table       (table root)
+                                   :url         (format "/%s/%s"
+                                                        (url root)
+                                                        (:rule rule)
+                                                        (:rule indepth))})))
+                        (take max-related))]
+       (log/info (format "Applying heuristic %s to %s."
+                         (:rule rule)
+                         (full-name root)))
+       (log/info (format "Dimensions bindings:\n%s"
+                         (->> dashboard
+                              :context
+                              :dimensions
+                              (m/map-vals #(update % :matches (partial map :name)))
+                              u/pprint-to-str)))
+       (log/info (format "Using definitions:\nMetrics:\n%s\nFilters:\n%s"
+                         (-> dashboard :context :metrics u/pprint-to-str)
+                         (-> dashboard :context :filters u/pprint-to-str)))
+       (-> dashboard
+           populate/create-dashboard
+           (assoc :related
+             {:tables  (->> root
+                            database
+                            candidate-tables
+                            (remove (comp #{root} :table))
+                            (take (- max-related (count indepth))))
+              :indepth indepth})))
+     (log/info (format "Skipping %s: no cards fully match the topology."
+                       (full-name root))))))
+
+(def ^:private ^{:arglists '([card])} table-like?
+  (comp empty? :aggregation :query :dataset_query))
+
+(defmulti
+  ^{:doc "Create a transient dashboard analyzing given entity."
+   :arglists '([entity])}
+  automagic-analysis type)
+
+(defmethod automagic-analysis (type Metric)
+  [metric]
+  (let [rule      (-> "special/metric.yaml"
+                      rules/load-rule
+                      (update :metrics conj {"Metric" {:metric ["METRIC" (:id metric)]
+                                                       :score  100}}))
+        context   (make-context (Table (:table_id metric)) rule)
+        cards     (make-cards context rule)
+        filters   (->> rule
+                       :dashboard_filters
+                       (mapcat (comp :matches (:dimensions context))))
+        dashboard {:title   (format "Analysis of %s" (:name metric))
+                   :groups  (:groups rule)
+                   :filters filters
+                   :cards   cards}]
+    (when cards
+      (populate/create-dashboard dashboard (count cards)))))
+
+(defmethod automagic-analysis (type Card)
+  [card]
+  (if (table-like? card)
+    (automagic-dashboard card)
+    nil))
+
+(defmethod automagic-analysis (type Query)
+  [query]
+  (if (table-like? query)
+    (automagic-dashboard query)
+    nil))
+
+(defmethod automagic-analysis (type Field)
+  [field]
```

Comment:
A comment around why this is empty would be good, maybe a `nil` too if that's what is intended